### PR TITLE
Allow for offline update of Airflow Releases

### PR DESCRIPTION
**Pull request by andriisoldatenko**
Resolves https://github.com/astronomer/issues/issues/2183

### CODE REVIEW

**Review by schnie — COMMENTED** [review]
@andriisoldatenko yes I think so. Would it be possible to provide these overrides as YAML instead of JSON though? Just because it'll flow nice with the rest of the platform config.

I was also thinking that we would leave the override configmap empty by default, so that out of the box the file baked into the image would be used. If the user actually provides a value to us, then we would mount this file into the Houston pod. Maybe we need some helm logic to conditionally mount this in, _only_ when it's provided by the user.

**Comment by andriisoldatenko** [comment]
> @andriisoldatenko would it be possible to provide these overrides as YAML instead of JSON though? Just because it'll flow nice with the rest of the platform config.
> 
> I was also thinking that we would leave the override configmap empty by default, so that out of the box the file baked into the image would be used. If the user actually provides a value to us, then we would mount this file into the Houston pod. Maybe we need some helm logic to conditionally mount this in, _only_ when it's provided by the user.

PTAL.

File: charts/astronomer/templates/houston/houston-configmap.yaml [hunk 1]
```diff
@@ -12,7 +12,11 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     heritage: {{ .Release.Service }}
 data:
+  {{ if .Values.airflowReleasesConfig }}
+  airflow_releases.json:
```
Comment (schnie) [inline charts/astronomer/templates/houston/houston-configmap.yaml]: @andriisoldatenko should we change this to yaml extension now?

Comment (andriisoldatenko) [reply charts/astronomer/templates/houston/houston-configmap.yaml]: @schnie i fixed code, PTAL

**Comment by andriisoldatenko** [comment]
@schnie 
practically LGTM

```
➜  astronomer git:(allow-for-offline-update-of-airflow-releases) ✗ k get configmaps astronomer-houston-config -o yaml          apiVersion: v1
data:
  airflow_releases.json: |
    {"available_releases":[{"channel":"stable","level":"new_feature","release_date":"2020-09-29T22:15:00+00:00","tags":["2.0.0-alpine3.10-onbuild"],"url":"https://github.com/astronomer/airflow/releases/tag/v1.10.12%2Bastro.1","version":"2.0.0"}]}
  local-production.yaml: |
    auth:
      local:
        enabled: true
 


/houston $ yarn check-airflow-updates --url https://111updates.astronomer.io/astronomer-certified2222
Explore the full API: http://pris.ly/d/client
info The preview flags `atomicNumberOperations` were removed, you can now safely remove them from your schema.prisma.
2020-12-07T16:09:14 INFO Starting update platform versions
2020-12-07T16:09:14 DEBUG Fetching airflow updates from https://111updates.astronomer.io/astronomer-certified2222
2020-12-07T16:09:14 ERROR Error: getaddrinfo ENOTFOUND 111updates.astronomer.io
2020-12-07T16:09:14 WARN Fallback to local file airflow_releases.json
2020-12-07T16:09:14 INFO Checking possible new release version 2.0.0
2020-12-07T16:09:14 INFO Checking possible new release tag 2.0.0-alpine3.10-onbuild
Done in 6.32s.


```

UI in local cluster:

image

File: charts/astronomer/templates/houston/houston-configmap.yaml (hunk 1 above)
Comment (andriisoldatenko) [reply charts/astronomer/templates/houston/houston-configmap.yaml]: I'm trying to respect original file from https://github.com/astronomer/houston-api/blob/main/airflow_releases.json,
so we define `airflowReleasesConfig` and than convert to json via `toJson`
```
data:
  {{ if .Values.houston.airflowReleasesConfig }}
  airflow_releases.json: |
    {{ .Values.houston.airflowReleasesConfig | toJson }}
  # These are system-specified config overrides.
  {{- end }}
```
and inside pod we can see correct file body and correct extension.
@schnie please correct me if i'm wrong.

Comment (andriisoldatenko) [reply charts/astronomer/templates/houston/houston-configmap.yaml]:
```$ kubectl exec -it astronomer-houston-5f7f765f4b-qzvdb -n astronomer sh
/houston $
/houston $ cat airflow_releases.json
{"available_releases":[{"channel":"stable","level":"new_feature","release_date":"2020-09-29T22:15:00+00:00","tags":["2.0.0-alpine3.10-onbuild"],"url":"https://github.com/astronomer/airflow/releases/tag/v1.10.12%2Bastro.1","version":"2.0.0"}]}
/houston $
```

**Review by schnie — APPROVED** [review]

File: charts/astronomer/values.yaml [hunk 2]
```diff
@@ -191,6 +191,39 @@ houston:
 
     # Default here is to run at midnight every night
     schedule: "0 0 * * *"
+  
+  # this option allow to overrides airflow releases config
```
Comment (vishwas-astro) [inline charts/astronomer/values.yaml:195]: this option allows to override airflow releases config

Comment (andriisoldatenko) [reply charts/astronomer/values.yaml:195]: done.

**Review by jpweber — APPROVED** [review]

**Pull request merged**